docs/concepts.md

```diff
@@ -0,0 +1,136 @@
+</pl-question-panel>
+
+<pl-number-input answers-name="y" label="$y =$"></pl-number-input>
+```
+
+#### Example `server.py`
+
+```python title="server.py"
+def generate(data):
+    data["params"]["x"] = random.randint(5, 10)
+    data["params"]["operation"] = random.choice(["double", "triple"])
+
+    if data["params"]["operation"] == "double":
+        data["correct_answers"]["y"] = 2 * data["params"]["x"]
+    else:
+        data["correct_answers"]["y"] = 3 * data["params"]["x"]
+```
+
+??? example "Question with custom partial credit & feedback"
+
+    === "Example `question.html`"
+
+        ```html title="question.html"
+        <pl-question-panel>
+            If $x = {{params.x}}$ and $y$ is {{params.operation}} $x$, what is $y$?
+        </pl-question-panel>
+
+        <pl-number-input answers-name="y" label="$y =$"></pl-number-input>
+        <pl-submission-panel> {{feedback.y}} </pl-submission-panel>
+        ```
+
+    === "Example `server.py`"
+
+        ```python title="server.py"
+        def generate(data):
+            data["params"]["x"] = random.randint(5, 10)
+            data["params"]["operation"] = random.choice(["double", "triple"])
+
+            if data["params"]["operation"] == "double":
+                data["correct_answers"]["y"] = 2 * data["params"]["x"]
+            else:
+                data["correct_answers"]["y"] = 3 * data["params"]["x"]
+
+        def parse(data):
+            if "y" not in data["format_errors"] and data["submitted_answers"]["y"] < 0:
+                data["format_errors"]["y"] = "Negative numbers are not allowed"
+
+        def grade(data):
+            if math.isclose(data["score"], 0.0) and data["submitted_answers"]["y"] > data["params"]["x"]:
+                data["partial_scores"]["y"]["score"] = 0.5
+                data["score"] = 0.5
+                data["feedback"]["y"] = "Your value for $y$ is larger than $x$, but incorrect."
+        ```
+
+### Elements
+
+Questions can use **elements** to accept student input, display diagrams, control how questions are displayed. Elements are pre-built interface elements for common questions, like numerical inputs, multiple choice, and more. If you require more flexibility -- **custom elements** are reusable components that can be tailored to your course and provide a way to create more complex questions.
+
+=== "Integer input element"
+
+    ![](elements/pl-integer-input.png)
+
+=== "HTML for element"
+
+    ```html title="question.html"
+    <pl-integer-input answers-name="int_value" label="$y =$"></pl-integer-input>
+    ```
+
+You can view a list of all the available elements in the [elements documentation](./elements.md).
+
+### Assessments
+
+A course contains a collection of questions, which are composed together to create [**assessments**](./assessment/index.md). Each course instance contains a collection of assessments. There are **exam assessments** and **homework assessments**.
```

Comment:
Do we need to discuss types here? This is actually somewhat confusing for beginners, and we have plans to eliminate the distinction.

```suggestion
A course contains a collection of questions, which are composed together to create [**assessments**](./assessment/index.md).
```

Reply:
> If we can give a quick summary ("homeworks give unlimited retries with different variants for mastery learning, exams give limited tries for summative assessments"), I'm open to this.

Reply:
> A course contains a collection of questions, which are composed together to create [**assessments**](./assessment/index.md). Each course instance contains a collection of assessments. There are **exam assessments** and **homework assessments**. Homeworks are formative assessments designed for mastery learning: they give unlimited retries with different variants, and incentivize repeated correct answers. Exams are summative assessments designed for measure learning: they give limited attempts with the same variant, and incentivize getting the correct answer on the first try.

Reply:
^ the above text was added.